json_rpc/jsonmarshal.nim

```diff
@@ -62,8 +69,11 @@ proc fromJson(n: JsonNode, argName: string, result: var bool) =
   result = n.getBool()
 
 proc fromJson(n: JsonNode, argName: string, result: var int) =
-  n.kind.expect(JInt, argName)
-  result = n.getInt()
+  if n.kind == JString:
```

Comment:
Is there a reason why you need to do this conversion here? I think web3.utils generally is meant for this work.

While I don't see an immediate risk in this case, I believe it would be beneficial if we keep these `fromJson` calls as simple as possible, and that would probably mean accepting only 1 type in general.

Reply:
Well in this particular case we could just change the method signatures from using `int` to `string`, not a big deal.

Reply:
I had a further look at `parseHexInt`, turns out it does not check for overflows. 
I didn't check in detail but I guess values such as `0x8000000000000000` and bigger will wrap around, not the behaviour you would want. (or at least different behaviour from providing currently an int that is too large)